File: ClientApp/src/app/shared/Models/tabel.ts
import { JsonApiModelConfig, JsonApiModel, Attribute, HasMany, BelongsTo } from 'angular2-jsonapi';
import { BaseTitleObject } from './_baseTitleObject';
import { Person } from './person';
import { Company } from './company';
import { Serviceline } from './serviceline';

@JsonApiModelConfig({ type: 'tabels' })
export class Tabel extends BaseTitleObject {
  @Attribute()  year: number;
  @Attribute()  month: number;
  @Attribute()  fill: boolean;
  @Attribute()  approve: boolean;
  @Attribute()  date: Date;
  @Attribute()  fillDate?: Date;
  @Attribute()  approveDate?: Date;

  @BelongsTo()  person: Person;
  @BelongsTo()  approver: Person;
  @BelongsTo()  company: Company;

  @HasMany()  services: Serviceline[];

  get strfill() { return this.getBooleanText(this.fill); }
  get strapprove() { return this.getBooleanText(this.approve); }
  get strfillDate() {
    return this.dPipe.transform(this.fillDate, this.dFormatShort);
  }
  get strapproveDate() {
    return this.dPipe.transform(this.approveDate, this.dFormatShort);
  }
  get strfillDatePicker() {
    return this.dPipe.transform(this.fillDate, this.dFormatShortDatePicker);
  }
  get strapproveDatePicker() {
    return this.dPipe.transform(this.approveDate, this.dFormatShortDatePicker);
  }
  // @HasMany()
  // comments: Comment[];
}

File: Models/DB/Tabel.cs
using JsonApiDotNetCore.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace SURV.Models.DB
{
    public class Tabel : BaseTitleObject
    {
        [Attr("year")]
        public int Year { get; set; }

        [Attr("month")]
        public int Month { get; set; }

        [Attr("fill")]
        [Column(TypeName = "BOOLEAN")]
        [DefaultValue(false)]
        public bool Fill { get; set; }

        [Attr("approve")]
        [Column(TypeName = "BOOLEAN")]
        [DefaultValue(false)]
        public bool Approve { get; set; }

        [Column(TypeName = "DATETIME")]
        [Attr("fillDate")]
        public DateTime? FillDate { get; set; }

        [Column(TypeName = "DATETIME")]
        [Attr("approveDate")]
        public DateTime? ApproveDate { get; set; }

        [ForeignKey(nameof(Approver))] public int? ApproverId { get; set; }

        [HasOne] public virtual Person Approver { get; set; }

        [ForeignKey(nameof(Company))] public int? CompanyId { get; set; }

        [HasOne] public virtual Company Company { get; set; }

        [ForeignKey(nameof(Person))] public int? PersonId { get; set; }

        [HasOne] public virtual Person Person { get; set; }

        [HasMany] public virtual List<Serviceline> Services { get; set; }
    }
}
